x/upgrade/client/cli/query.go

```diff
@@ -1,98 +1,105 @@
 package cli
 
 import (
-	"encoding/binary"
+	"context"
 	"fmt"
 
-	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
-
 	"github.com/spf13/cobra"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
 )
 
-// GetPlanCmd returns the query upgrade plan command
-func GetPlanCmd(storeName string, cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
+// GetQueryCmd returns the parent command for all x/upgrade CLi query commands
+func GetQueryCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   types.ModuleName,
+		Short: "Querying commands for the upgrade module",
+	}
+	cmd.AddCommand(
+		GetCurrentPlanCmd(),
+		GetAppliedPlanCmd(),
+	)
+
+	return cmd
+}
+
+// GetCurrentPlanCmd returns the query upgrade plan command
+func GetCurrentPlanCmd() *cobra.Command {
+	cmd := &cobra.Command{
 		Use:   "plan",
 		Short: "get upgrade plan (if one exists)",
 		Long:  "Gets the currently scheduled upgrade plan, if one exists",
 		Args:  cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx := client.NewContext().WithCodec(cdc).WithJSONMarshaler(cdc)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			queryClient := types.NewQueryClient(clientCtx.Init())
 
-			// ignore height for now
-			res, _, err := clientCtx.Query(fmt.Sprintf("custom/%s/%s", types.QuerierKey, types.QueryCurrent))
+			params := types.NewQueryCurrentPlanRequest()
+			res, err := queryClient.CurrentPlan(context.Background(), params)
 			if err != nil {
 				return err
 			}
 
-			if len(res) == 0 {
+			if len(res.Plan.Name) == 0 {
 				return fmt.Errorf("no upgrade scheduled")
 			}
 
-			var plan types.Plan
-			err = cdc.UnmarshalJSON(res, &plan)
 			if err != nil {
 				return err
 			}
-			return clientCtx.PrintOutput(plan)
+			return clientCtx.PrintOutput(res.Plan)
 		},
 	}
+
+	return flags.GetCommands(cmd)[0]
 }
 
-// GetAppliedHeightCmd returns the height at which a completed upgrade was applied
-func GetAppliedHeightCmd(storeName string, cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
+// GetAppliedPlanCmd returns the height at which a completed upgrade was applied
```

Comment:
We might need to update this doc, its returning entire blockinfo not just `height` right?